### PR TITLE
turn view to a function when layout exists

### PR DESCRIPTION
**Pull request by xiangnanscu**

### CODE REVIEW

File: lib/resty/template.lua [hunk 1]
```diff
@@ -291,15 +291,15 @@ local function new(template, safe)
                 render = function(self, context)
                     context = context or self
                     context.blocks = context.blocks or {}
-                    context.view = template.process(view, context)
+                    context.view = function() return template.process(view, context) end
```
Comment (bungle) [inline lib/resty/template.lua:294]: Should we make it possible to override the `context`: `{* view { foo = "bar" } *}`?
```suggestion
                    context.view = function(ctx) return template.process(view, ctx, context) end
```

**Comment by xiangnanscu** [comment]
hello. Should the other three places need to apply this change?

**Comment by bungle** [comment]
> hello. Should the other three places need to apply this change?

Yes, all three.

**Comment by xiangnanscu** [comment]
@bungle done

File: lib/resty/template.lua [hunk 2]
```diff
@@ -308,13 +308,13 @@ local function new(template, safe)
...
                     context.blocks = context.blocks or {}
-                    context.view = template.process(view, context)
+                    context.view = function(ctx) return template.process(view, ctx or context) end
                     template.render(layout, context)
                 end
                 process = function(self, context)
                     context = context or self
                     context.blocks = context.blocks or {}
-                    context.view = template.process(view, context)
+                    context.view = function() return template.process(view, context) end
```
Comment (bungle) [inline lib/resty/template.lua:317]: Is it still missing here?

**Review by bungle — CHANGES_REQUESTED** [review]

**Comment by xiangnanscu** [comment]
@bungle sorry. done

**Pull request merged**